src/rpc/client.zig

```diff
@@ -344,6 +344,25 @@ pub const Client = struct {
         return self.sendFetchRequest(allocator, types.RpcVersionInfo, request, .{});
     }
 
+    const GetVoteAccountsConfig = struct {
```

Comment:
The whole point of the rpc client is to convert between strings and types. Otherwise the user might as well just use json directly with a plain old http client. Not using `Pubkey` here just seems to make client less useful, without any benefit that I can see.

I would be careful about where the types come from though. The RPC client should probably not depend on anything outside of `core`. Depending on plain old data types seems that define the general domain of Solana seems completely fine to me. But depending on other components like the ledger which depends on rocksdb, that's where you start running into actual costs.

Reply:
In my mind the whole purpose of the rpc client is to provide a simple api for making rpc requests. As far as requests go I don't think the above example makes to client any less useful since it is just a couple of extra characters. For responses I also prefer to selectively parse into sig types when needed rather than parsing everything by default. At the end of the day though I don't really mind so if you want you can just go ahead and resolve this comment.

Reply:
> I don't really mind so if you want you can just go ahead and resolve this comment.

I appreciate this, but I'd like to get on the same page about how the rpc client should be designed overall. I still don't follow your reasoning. If you prefer, we can continue this discussion in another medium.

> the whole purpose of the rpc client is to provide a simple api for making rpc requests

But what does this mean? In this context, "providing an API" means you're allowing the user to work with pre-defined structs instead of strings. There's not much else going on here other than sending strings with an http client. Imagine if we removed all type conversions from this client. Then the rpc client is barely distinguishable from a generic http client. In that case, there would not be much of a point in having an rpc client library.

> I don't think the above example makes to client any less useful since it is just a couple of extra characters.

I could extend the same argument to `GetAccountInfoConfig`. Why don't we eliminate that struct and have the user pass a json string instead? It's just a couple extra characters. In my mind, using `Pubkey` follows the same logic as using the other structs used by the rpc client. If you're going to provide an rpc client that does the type conversations, why stop halfway? Why not convert everything into safer and more useful types?